packages/git/src/browser/git-widget.tsx

```diff
@@ -194,6 +284,35 @@ export class GitWidget extends GitDiffWidget implements StatefulWidget {
         }
     }
 
+    protected async unamend(): Promise<void> {
+        if (this.transition.state !== 'none' && this.transitionHint !== 'none') {
+            return;
+        }
+
+        const commitToRestore = (this.amendingCommits.length >= 1)
+            ? this.amendingCommits[this.amendingCommits.length - 1]
+            : undefined;
+        const oldestAmendCommit = (this.amendingCommits.length >= 2)
+            ? this.amendingCommits[this.amendingCommits.length - 2]
+            : undefined;
+
+        const { selectedRepository } = this.repositoryProvider;
+        if (selectedRepository && commitToRestore) {
+            const message = oldestAmendCommit
```

Comment:
This logic is very similar to the `amend` logic, can we extract and reuse it? If we ever hit a bug, we can fix it one place ;)